### PR TITLE
TestCafe wait for results

### PR DESCRIPTION
**Pull request by yen-tt**
Update cafeTest acceptance test

- Added `waitOnSearchComplete` in verticalResults.js to wait till query responses return status 200 and searchComplete state appears on page.
- Register requestLogger to each test to track vertical request and response to Answers API endpoint
- Added IE11nocachehook from SDK
- From oliver's acceptance test failure investigation..
&ensp; - first point: the test first function call is `verticalResults.scrollToBottom()`, which wait on `this._resultsWrapper` when perform scroll on page. This resultsWrapper will always exist because it's set in our page template. The test didn't account for waiting on the actual search result div. Calling `waitOnSearchComplete` before scrolling would prevent this issue.
&ensp; - second point: removed the arbitrary wait time 1s in `dragLeft()`. Replaced with waiting for the response to be returned first using searchComplete state.
- Updated tests to wait for results before calling `getNumResults()` which check if results exists after a search. (From testCafe documentation, Selector wait/timeouts functionality have no effect on Selector.exists and Selector.count properties. These properties are evaluated immediately.)

J=SLAP-1425
TEST= auto

ran testCafe locally to verify tests and wait time in selectors

### CODE REVIEW

File: tests/acceptance/blocks/verticalresults.js
Comment (tmeyer2115) [inline tests/acceptance/blocks/verticalresults.js:73]: Proper capitalization

File: tests/acceptance/blocks/verticalresults.js
Comment (tmeyer2115) [inline tests/acceptance/blocks/verticalresults.js:77]: For my own edification, what is `preSearch`?

File: tests/acceptance/blocks/verticalresults.js
Comment (tmeyer2115) [inline tests/acceptance/blocks/verticalresults.js:82]: Does `await this._searchComplete()` being successful imply `t.expect(this._searchComplete.exists).ok()` automatically?

File: tests/acceptance/blocks/verticalresults.js
Comment (tmeyer2115) [inline tests/acceptance/blocks/verticalresults.js:76]: If someone calls this method, but no search has actually been executed, will it timeout? I'm wondering if we should handle that situation gracefully, like it returns immediately. Maybe not something we need to worry about, just a thought.

File: tests/acceptance/blocks/verticalresults.js
Comment (yen-tt) [reply tests/acceptance/blocks/verticalresults.js:86]: test would still consider passed if `await this._searchComplete()` doesn't find any matching element

File: tests/acceptance/blocks/verticalresults.js
Comment (yen-tt) [reply tests/acceptance/blocks/verticalresults.js]: it's one of the search states from SDK (presearch, loading, completed). I originally thought initial search on page may go from presearch to completed, skipping loading state. Turns out it's just testCafe unable to catch the loading transition. Updated to track query request/responses instead!

File: tests/acceptance/blocks/verticalresults.js
Comment (yen-tt) [reply tests/acceptance/blocks/verticalresults.js]: in this method, and in the updated one, it will timeout eventually. I think if no search is executed and someone uses this method, meaning they are expecting a search or result returned of some sort, it make sense to indicate a failure here to notify that the expected behavior is not met

File: tests/acceptance/blocks/verticalresults.js
Comment (cea2aj) [inline tests/acceptance/blocks/verticalresults.js:80]: Is this code waiting while an invalid response exists? is there a reason for that approach rather than waiting until a successful response has been logged?

Comment (yen-tt) [reply tests/acceptance/blocks/verticalresults.js]: Logger will hold all request/response throughout a test so just checking for a successful response might accidentally check on previous requests. But, after some thoughts, I updated this method check for a successful response in the loop, and clear out the logger at the end of this method. This make sure we are always verifying the latest response.

File: tests/acceptance/suites/vertical-full-page-map-with-filters.js
Comment (oshi97) [inline tests/acceptance/suites/vertical-full-page-map-with-filters.js:12]: should the ie11 hook go inside registerLogger? like, is there ever a time we would call registerLogger without also registering the ie11 hook?

Comment (yen-tt) [reply tests/acceptance/suites/vertical-full-page-map-with-filters.js]: yeah, that would make sense. Updated!

**Review by oshi97 — APPROVED** [review]

File: tests/acceptance/blocks/verticalresults.js
Comment (oshi97) [inline tests/acceptance/blocks/verticalresults.js:76]: Sorry I know I just recently approved this, but after looking at the SDK half of this item maybe this waitOnSearchComplete should go into its own request logger class instead of inside the VerticalResults component's class?

Comment (cea2aj) [reply tests/acceptance/blocks/verticalresults.js:76]: I agree, if we can move this to it's own class we could easily expand it to universal search.  Yen, if we have a new class such as `SearchRequestLogger`, we could have a `registerVerticalSearchLogger` and a `registerUniversalSearchLogger` function so that we could set it up for either type of search

**Review by oshi97 — APPROVED** [review]

**Pull request merged**
